### ISSUE TITLE
rsync with --exclude-paths results in error output due to core-status --project argument issue

### ISSUE THREAD

**Issue by swichers** [drush-ops/drush#1291]
If you use an --exclude-paths options as such:

     drush rsync @remote:%files @local:%files --exclude-paths=:styles:xmlsitemap:css:js:cdn:ctools

Then you will get an error:

    The external command could not be executed due to an application error.

The execution of the command completes normally after several seconds of delay (I assume drush is waiting for a process to time out?)

If I run the same drush rsync command with the -d flag then I can see that the command which is failing is:

    /usr/local/bin/php  -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/local/Cellar/drush/6.5.0/libexec/drush.php --php=/usr/local/bin/php --php-options=' -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off'  --backend=2 --verbose --debug --root=/path/to/project core-status   --project=files:styles:xmlsitemap:css:js:cdn:ctools 2>&1

You'll note that my --exclude-path options were put into the --project argument. As a side note, it also added "files" to my exclude list and I can't find a way to stop that. If I run that command directly then the error I wind up seeing is:

    PHP Fatal error:  Call to undefined function drupal_system_listing() in /path/to/project/modules/system/system.module on line 2364
    
If I remove the --project argument but run the above otherwise unchanged, then drush completes normally. 

It seems to me like --exclude-paths is erroneously being used to populate --project.

Here is the relevant log from executing drush with the debug flag.

```
Bootstrap to phase 0. [0.01 sec, 1.88 MB]                            [bootstrap]
Drush bootstrap phase : _drush_bootstrap_drush() [0.01 sec, 2 MB]    [bootstrap]
Cache MISS cid: 6.5.0-alias-path--6a320157824c25fef1db517da1e0f523 [0.01 sec, 2.01 MB]                  [debug]
Loading drushrc "/path/to/project/sites/all/drush/drushrc.php" into "drupal" scope. [0.01 sec, 2.01 MB]                 [bootstrap]
Cache HIT cid: 6.5.0-commandfiles-0-1f515e22f154439c3840684f85cbbd33 [0.02 sec, 2.03 MB]                  [debug]
Bootstrap to phase 0. [0.15 sec, 6.07 MB]                 [bootstrap]
Bootstrap to phase 0. [0.18 sec, 6.07 MB]                 [bootstrap]
Found command: core-rsync (commandfile=core) [0.18 sec, 6.07 MB]                  [bootstrap]
Including /usr/local/Cellar/drush/6.5.0/libexec/commands/core/rsync.core.inc [0.28 sec, 6.08 MB]                  [bootstrap]
Calling hook drush_core_rsync [0.28 sec, 6.14 MB]                 [debug]
Loaded alias @remote from file /Users/steven.wichers/.drush/remote.aliases.drushrc.php [0.4 sec, 6.15 MB]                  [notice]
Loaded alias @local from file /Users/steven.wichers/.drush/local.aliases.drushrc.php [0.58 sec, 6.19 MB]                                                                                                                                                    [notice]
Cache MISS cid: 6.5.0-alias-path--6a320157824c25fef1db517da1e0f523 [0.71 sec, 6.21                  MB]                 [debug]
Undefined index: uri sitealias.inc:928 [0.71 sec, 6.21 MB]                  [notice]
Backend invoke: /usr/local/bin/php  -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/local/Cellar/drush/6.5.0/libexec/drush.php --php=/usr/local/bin/php --php-options=' -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off'  --backend=2 --verbose --debug --root=/path/to/project  core-status   --project=files:styles:xmlsitemap:css:js:cdn:ctools 2>&1 [0.84 sec, 6.22 MB]   [command]
/usr/local/bin/php  -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/local/Cellar/drush/6.5.0/libexec/drush.php --php=/usr/local/bin/php --php-options=' -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off'  --backend=2 --verbose --debug --root=/path/to/project  core-status   --project=files:styles:xmlsitemap:css:js:cdn:ctools 2>&1 [0.84 sec, 6.22 MB]                    [notice]
The external command could not be executed due to an application error. [61.11 sec, 6.24 MB]                [error]
(normal output after)
```

Here is the relevant log from executing the command directly with the bad --project option.

```
Bootstrap to phase 0. [0.01 sec, 1.88 MB]                            [bootstrap]
Drush bootstrap phase : _drush_bootstrap_drush() [0.01 sec, 2 MB]    [bootstrap]
Cache MISS cid: 6.5.0-alias-path--6a320157824c25fef1db517da1e0f523 [0.02 sec, 2.01 MB]              [debug]
Loading drushrc "/path/to/project/sites/all/drush/drushrc.php" into "drupal" scope. [0.02 sec, 2.01 MB]         [bootstrap]
Cache HIT cid: 6.5.0-commandfiles-0-1f515e22f154439c3840684f85cbbd33 [0.04 sec, 2.02 MB]              [debug]
DRUSH_BACKEND:{"type":"bootstrap","message":"Bootstrap to phase 0.","timestamp":1427417110.1046,"memory":6360312,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"bootstrap","message":"Drush bootstrap phase : _drush_bootstrap_drupal_root()","timestamp":1427417110.148,"memory":7338984,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"notice","message":"Initialized Drupal 7.34 root directory at \/path\/to\/project","timestamp":1427417110.1539,"memory":7341056,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"bootstrap","message":"Drush bootstrap phase : _drush_bootstrap_drupal_site()","timestamp":1427417110.1544,"memory":7347616,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"notice","message":"Initialized Drupal site default at sites\/default","timestamp":1427417110.1545,"memory":7349464,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"debug","message":"Cache HIT cid: 6.5.0-commandfiles-2-062e26fa8ebdd8c444d688ab9447c986","timestamp":1427417110.1606,"memory":7352496,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"bootstrap","message":"Drush bootstrap phase : _drush_bootstrap_drupal_configuration()","timestamp":1427417110.1612,"memory":7351168,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"bootstrap","message":"SQLSTATE[HY000] [2002] Operation timed out","timestamp":1427417170.1671,"memory":7397464,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"bootstrap","message":"Found command: core-status (commandfile=core)","timestamp":1427417170.1674,"memory":7385672,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"notice","message":"Loading outputformat engine.","timestamp":1427417170.2377,"memory":7421248,"error":null,"packet":"log"}
DRUSH_BACKEND:{"type":"debug","message":"Calling hook drush_core_status","timestamp":1427417170.2394,"memory":7473240,"error":null,"packet":"log"}
PHP Fatal error:  Call to undefined function drupal_system_listing() in /path/to/project/modules/system/system.module on line 2364

Fatal error: Call to undefined function drupal_system_listing() in /path/to/project/modules/system/system.module on line 2364
(normal output after)
```

**Comment by swichers** [comment]
I can say that changing from using --exclude-paths to listing individual --exclude options does not trigger the same problem.

**Comment by weitzman** [comment]
For starters, please retry using the new 6.6.0. If that fails, please try master branch or 7.0.0-rc1.

**Comment by swichers** [comment]
I can't even duplicate this on my original setup now. Either the steps I was using to duplicate the issue aren't the real reason why the issue occurs, or I had a tainted environment during my discovery and testing. I'll continue to try and duplicate this, but I have a script that would consistently generate the issue and now it doesn't.

**Comment by swichers** [comment]
Okay, I figured out the missing step. 

If the database server is not available when using 6.5 then you will get that error. If the database server is available then the error does not appear.

I can confirm that 6.6 still has the issue and that the latest version of 7 does not.

I'm not sure if this is a supported use case (using drush rsync on a Drupal install with no DB currently available). It sounds to me like it wouldn't be, but I could see the argument around the initial site setup.
